app/views/admin/registrations/index.pdf.prawn

```diff
@@ -23,10 +23,8 @@ prawn_document(force_download: true, filename: @pdf_filename, page_layout: :land
     row << registration.departure.to_s || ''
 
     @conference.questions.each do |question|
-      qa = registration.qanswers.find_by(question: question)
-      answer = ( qa ? qa.answer.title : '' )
-
-      row << answer
+      qa = registration.qanswers.find{|qa| qa.question_id == question.id }
```

Comment:
why is that better than `qa = registration.qanswers.find_by(question: question)`? :thinking:

Reply:
`ActiveRelation#find_by` peforms an additonal query against the database, for something that can be solved much faster in Ruby.

Reply:
database queries are normally much faster than manipulating Ruby objects... :confused: 

`registration.qanswers.find{|qa| qa.question_id == question.id }` will also triger a query to the database

```
> registration = Registration.last
  Registration Load (0.1ms)  SELECT  "registrations".* FROM "registrations" ORDER BY "registrations"."id" DESC LIMIT ?  [["LIMIT", 1]]
=> #<Registration id: 14433, conference_id: 1, arrival: nil, departure: nil, created_at: "2018-06-15 12:57:31", updated_at: "2018-06-15 12:57:31", other_special_needs: nil, attended: false, volunteer: nil, user_id: 14437, week: 24, accepted_code_of_conduct: nil>

> registration.qanswers.find{|qa| qa.question_id == 2 }
  Qanswer Load (0.1ms)  SELECT "qanswers".* FROM "qanswers" INNER JOIN "qanswers_registrations" ON "qanswers"."id" = "qanswers_registrations"."qanswer_id" WHERE "qanswers_registrations"."registration_id" = ?  [["registration_id", 14433]]

> registration.qanswers.find(2)
  Qanswer Load (6.7ms)  SELECT  "qanswers".* FROM "qanswers" INNER JOIN "qanswers_registrations" ON "qanswers"."id" = "qanswers_registrations"."qanswer_id" WHERE "qanswers_registrations"."registration_id" = ? AND "qanswers"."id" = ? LIMIT ?  [["registration_id", 14433], ["id", 2], ["LIMIT", 1]]
```

Reply:
That depends on latency to your database. Osem is loaded with inefficient queries that are masked by high database response rate.

For LFNW, our database is disparate from our application server; the latency "lowers the water" to show the rocks that impede flow.

Strategies like this have been employed on other problem areas, especially the splashpage, resulting in > 30x performance improvement overall; that performance benefits everyone (even if you don't see it in your query response time) by generating less waste and utilizing limited resources more efficiently.